integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -29,6 +29,25 @@
 
 pytestmark = pytest.mark.nightly_resource_consuming_test
 
+# Choose the right Parquet conf keys.
+# The 'spark.sql.legacy.parquet.*' conf keys were removed completely on Apache Spark 4.0.
+# This change was pulled into Databricks 14.3.  For Spark versions that exceed either of
+# these, using the removed conf keys would cause deprecation errors, indicating that the
+# replacement conf keys (i.e. 'spark.sql.parquet.*') be used instead.
+must_use_new_parquet_conf_keys = is_spark_400_or_later() or is_databricks_version_or_later(14, 3)
+conf_key_parquet_datetimeRebaseModeInWrite = \
+  'spark.sql.parquet.datetimeRebaseModeInWrite' if must_use_new_parquet_conf_keys else \
```

Comment:
This and possibly others were added in Spark 3.2.0. Any reason we can't use this regardless of the version?

Reply:
I thought about just moving it to the new conf name, and be done.

The issue we might not land up covering is that our users might be using the old conf name.  We've seen these deprecated confs removed in pre-4.0 versions.  While it's improbably, we might want to catch the case where these are removed in older versions, maybe?

Reply:
Actually, nix that.    

Let me check the viability of moving to the new conf names.

Reply:
I've switched to the new confs.  The `spark-rapids` plugin does not have explicit references to this conf, so we should be able to do either.

I don't think we lose test coverage by using the non-deprecated conf version.  Any users using the deprecated conf on a Spark version where it isn't supported isn't likely to get off the ground in the first place.